### DISCUSSION TITLE
Initialize proxy device on home assistant issue

### DISCUSSION THREAD

**Discussion by lukaszlk** [s-allius/tsun-gen3-proxy#263] [Q&A] [answered]
Hi Stefan @s-allius,  perhaps I could seek your help here.

I own two MS-1800 micro inverters  so they are GEN3PLUS devices. I am struggling with getting everything to work.

Firstly, I've looped a connection from iot.talent-monitor.com to the IP of my home assistant device, as seen below, I've used your manual except I didn't configure Rebind \protection since I'm using a TP link set up.

![IMG_0829](https://github.com/user-attachments/assets/f17cd6e6-7a42-4a10-9993-63614aa5d11f)

After doing that, the micro inverters are no longer sending any status to Tsun Smart App.

![IMG_0831](https://github.com/user-attachments/assets/914547f8-723c-4a1f-b9b4-bc31f99d58c0)

I've configured the add-on as below:

 serial: Y17E7A0E060XXXX
  monitor_sn: "206828XXXX"
  node_id: PV-Garage
  suggested_area: Garage
  modbus_polling: false
  client_mode.host: 192.168.0.X
  client_mode.port: 8899
  client_mode.forward: true
  
And in log it returns:'2024-12-23 16:34:38  INFO | conn | 'client_mode' for snr: 206828XXXX host: 192.168.0.X:8899, forward: True'

After that I am stucked on'2024-12-23 16:34:38  INFO | root | Initialize proxy device on home assistant'

Could you help me with where do I make a mistake?
When accessing my micro inverters directly using 10.10.100.254, there's no option for me to change Server A, I could only add Server B but even then I'm still stuck on initialize proxy device.
![IMG_0832](https://github.com/user-attachments/assets/04fa2936-9c9f-4dbe-8726-02f430db4489)

Maybe I am mixing container proxy config with a config required for an add-on, is there a way you can guide me step by step how to configure an add-on of my inverter? Thank you

**Comment by s-allius** [comment]
Hi lukaszik,

you properly have inverters which establishes an encrypted connection to the cloud over port 10443. So you must use the client_mode. For this you have to set the `client_mode.host: 192.168.0.X` to the inverters IP address. It is important that the inverter has a fixed IP or which you should configure in your Router /DHCP server.

If the inverter is running due to enough PV power, you can ping the inverter and the proxy will establish a connection automatically. If you have a firewall or different layer 3 networks you must allow the connection from the proxy to the inverter with the target port 8899.

 If this all, don't work you can install a debug version of the proxy to get more logs with more details. You may allow the `advanced` mode in HA for the add-ons. Otherwise you will only see the release versions in the repro and not the debug or dev versions.

**Reply by lukaszlk** [reply]
Thank you,

Yes, the are set to the IP of the inverters, i have two devices with adress reservation set in my router for both of them.
I tried runing Debugg/dev but every time it stops on „ Initialize proxy device on home assistant'” and doesnt go any further and I dont know what the cause might be. Could it be connected to the issue that even with Solarman cloud enabled when looping the connection to the HA device they then wont Connect to the cloud server? Wondering where I make a mistake.

**Reply by s-allius** [reply]
`Initialize proxy device on home assistant` is the last step of the proxy's setup phase, where the proxy has received the hello packet from your HA. After that the proxy is awaiting incoming connections on port 5005 (only GEN3) and 10000 (only GEN3PLUS unencrypted) and try to establish outgoing connections to the GEN3PLUS inverters on port 8899. If there is any response you will see this in the logs. 

This means that you inverter is offline or your router doesn't route or block the connection. Maybe you can make a router trace for port 8899 or you can make a Wireshark trace in your network, to find the root cause.

**Reply by s-allius** [reply]
The Solarman cloud enabled if relevant for connection from the inverter to the proxy over port 10000 (Server-Mode). It doesn't influence the connection between the inverter and the proxy. 
In the Client-Mode you have to variants. 

1. you don't touch the DNS configuration and the proxy send the measurement to the TSUN Cloud, even if the proxy requests measurements per client-mode.
2. you block the connection from the inverter to the TSUN Cloud. Then you can forward all measurements, which the proxy gets from the inverter. For this you must set `client_mode.forward: true`

**Reply by lukaszlk** [reply]
Thank you.

So to continue using the app do you sugest to stop looping the connection to iot.talent-monitoring.com to my HA device or did i misunderstood and you mean the connfig of the inverter?

**Reply by s-allius** [reply]
No you don't have to stop looping. You can use the app by forwarding the measurement. For this the proxy has an emulation mode and will register the proxy as your inverter at the TSUN Cloud. This has the advantage, that the TSUN Cloud can't reconfigure your inverter without you knowledge.
But first you have to find the connection problem between the proxy and the inverters. Could you make a trace?

**Reply by lukaszlk** [reply]
Yes i can do it, but will the results be correct even tho the inverters are offline right now? And should I trace 8899-inverter or 8899-HA device?

**Reply by s-allius** [reply]
If your inverter is offline, then there can't be established any connection. You have wait until the sun raised up.

The port 8899 is the well-know port of your inverter, on which he is listening for incoming connections. So you have to trace Inverter-IP:8899 as destination

**Reply by lukaszlk** [reply]
Thank you, will do it in the morning when both of my inverters will be online and I will come back with the results.

**Comment by lukaszlk** [comment]
Hi again Stefan @s-allius 

I just ran a trace route from inverter to 8899, results as below:

 % sudo tcptraceroute 192.168.0.240 8899

Selected device en0, address 192.168.0.50, port 49681 for outgoing packets
Tracing the path to 192.168.0.240 on TCP port 8899, 30 hops max
 1  * * *
 2  * *libnet_write failed?  Attempted to write 40 bytes, only wrote -1

even tho both inverters are online and connected to Tsun app and a wifi I cannot ping them

Maybe it's the way my internet is set up? I run a AX5400 as a main router-Tp link switch-3 Deco x50 nodes. Inverters are connected to a guest network provided by Deco nodes. The same way I cannot ping them I cannot access their web interface by their IP address, I have to connect directly to their WIFI and use 10.10.100.254 to do it.

**Comment by lukaszlk** [comment]
It looks like the issue was with my internet setup, when reconnected inverter to the main router it works. Thank you Stefan for your help.

**Reply by s-allius** [reply, marked as answer]
When you connect device to the guest network, they normally are isolated and can't see each other. And of course, device from the guest net should not be able to connect to any device in you local network. Guest network are normally configured to reach only the internet.
I don't know TP-link but I'm sure the do it also in that way.

But I got it right, that the proxy gets measurements from you inverters now?

**Reply by lukaszlk** [reply]
Yes, it Works just like you intendet it to, so Thank you. Btw its an awesome addon, Thank you for your hard work, much appreciated

**Comment by lukaszlk** [comment]
Hi Stefan @s-allius, I tought i wont open a new discussion. I've managed to connect to my 2 inverters, they are detected as two, but even tho they display different SN, they display all the same values- production daily, total production, temperature etc. Looks like both of them are controlled by one controller which assigns an IP of only one of inverters, which causes both of them to display the same values of one of them. Should I fix something to differentiate them more or do you know what is causing the issue?

**Reply by s-allius** [reply]
Hi lukaszlk,

maybe there is something broken in your configuration. You should have to inverter blocks in your config. And each block must have a dedicated `monitor_sn`, `node_id` and `client_mode.host`. Is it possible that both inverters habe the same `node_id`?

It might be helpful to inspect you proxy.log file, if there are two inverters active

**Reply by lukaszlk** [reply]
HA reboot helped, but the only thing now is, when adding a Total Daily production from both inwerters to energy Dashboard, it shows strange values- daily production from both of them is around 0.5kWh and in a Dashboard it says that hourly production of 1 piece is around 1.5kWh so I dont know what that value is. Did I Add wrong ones as a production to the Dashboard?

**Comment by s-allius** [comment]
Until now I used a Shelly to measure the energy for the EnergieBoard. I just added the "Tota Energie" of my both inverters to the energy board and got this:
![C76F11BB-B3E2-4713-A796-20DB046405B7](https://github.com/user-attachments/assets/d4ee062f-59ea-44b0-8e85-ec1df1692b93)

Of course there are now three energy meters, so I got the doubled value. I made this to see if there are diffs between the Energie delta of the inverters compared to the value of the Shelly. For correct values I must remove the Shelly or the two inverters.

In you screenshot I also see three Energie values, which might be wrong if you have only two inverters. The next point is, that the energy board make calculations on the values. So it is important that you use the TotalEnergie value and not the Daily Energie values.
The Daily counter will be reseted each night at midnight. So there can be added an error cause of this. 

So best way is to use the Total one, this will give you the right sums even if you ha is down for a day and you miss some daily energy values.

**Reply by lukaszlk** [reply]
Hi,

The third one is a Hoymiles DTU since i have a mixed setup.Thats why I wanted to use HA as sort of set up that will give me a totall production. 
So, instead of Daily production to use total one for both correct?

**Reply by s-allius** [reply]
Yes, I made better experiences with the total counter.

**Reply by lukaszlk** [reply]
Thanks, will try it out. Thank you for your help.
